### PR TITLE
Add buff bonus list ui

### PR DESCRIPTION
**Pull request by tyrosine1153**
### Description

- Add buff bonus list UI to buff bonus popup

### Related Links

[스테이지 1회성 버프 폴리싱 - 버프 구매화면 고도화](https://app.asana.com/0/0/1202603085251652/f)

### Screenshot

bufflist

### CODE REVIEW

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs [hunk 1]
```diff
@@ -151,5 +173,44 @@ private void PushAction(bool advanced)
...
+                foreach (var randomBuffRow in randomBuffGroup)
+                {
+                    // buff
+                    go = Instantiate(buffPrefab, cellContainer);
+                    go.GetComponent<BuffBonusBuffCell>().Set(randomBuffRow);
+                }
+            }
+        }
+
+        private void InitializeChildren()
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:205]: Do you can explain about this method?
Sorry but I can't understand the relationship between 'Initializing' and 'Destruction'...
What is this destruction for?

Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:205]: The part that I forgot to modify when transferred the previous. I think it's better not to separate the method.

Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs]: How about this?

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs [hunk 2]
```diff
@@ -151,5 +173,39 @@ private void PushAction(bool advanced)
...
+
+            foreach (var cell in cellList)
+            {
+                Destroy(cell);
+            }
+
+            foreach (var randomBuffGroup in randomBuffGroupsByRank)
+            {
+                // Grade
+                var cell = Instantiate(titlePrefab, cellContainer);
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:199]: If you cached it when you Instantiate a cell, you can destroy it without calling GetComponentsInChildren()

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs [hunk 3]
```diff
@@ -151,5 +173,39 @@ private void PushAction(bool advanced)
...
+                cell.GetComponent<BuffBonusTitleCell>().Set(randomBuffGroup.Key);
+
+                foreach (var randomBuffRow in randomBuffGroup)
+                {
+                    // buff
+                    cell = Instantiate(buffPrefab, cellContainer);
```
Comment (hskim881028) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:205]: If you cached it when you Instantiate a cell, you can destroy it without calling GetComponentsInChildren()

**Review by hskim881028 — CHANGES_REQUESTED** [review]

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs (hunk 2 above)
Comment (tyrosine1153) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:198]: I cached the cells in new field. Thank you!

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs (hunk 1 above)
Comment (sonohoshi) [reply nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs]: Um... I still don't know why it destroy cells.

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs [hunk 4]
```diff
@@ -151,5 +171,35 @@ private void PushAction(bool advanced)
...
+        {
+            var rect = cellContainer.GetComponent<RectTransform>();
+            var pos = rect.anchoredPosition;
+            pos.y = 0;
+            rect.anchoredPosition = pos;
+
+            buffListView.SetActive(true);
+        }
+
+        private void initializeBuffListView()
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:185]:
```suggestion
        private void InitializeBuffListView()
```

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs [hunk 5]
```diff
@@ -27,6 +29,21 @@ public class BuffBonusPopup : PopupWidget
...
+        private GameObject buffListView = null;
+
+        [SerializeField]
+        private Transform cellContainer;
+
+        [SerializeField]
+        private GameObject titlePrefab;
+
+        [SerializeField]
+        private GameObject buffPrefab;
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:45]:
```suggestion
        private BuffBonusBuffCell buffPrefab;
```

If you change type of this prefab object, you would not need `GetComponent` in `initializeBuffListView()`.

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs [hunk 6]
```diff
@@ -27,6 +29,21 @@ public class BuffBonusPopup : PopupWidget
...
         private System.Action _onAttract;
 
+        [SerializeField]
+        private Button buffListButton = null;
+
+        [SerializeField]
+        private GameObject buffListView = null;
+
+        [SerializeField]
+        private Transform cellContainer;
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:39]:
```suggestion
        private RectTransform cellContainer;
```

File: nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs [hunk 7]
```diff
@@ -27,6 +29,21 @@ public class BuffBonusPopup : PopupWidget
...
+
+        [SerializeField]
+        private GameObject titlePrefab;
```
Comment (sonohoshi) [inline nekoyume/Assets/_Scripts/UI/Widget/Popup/BuffBonusPopup.cs:42]:
```suggestion
        private BuffBonusTitleCell titlePrefab;
```

**Review by sonohoshi — CHANGES_REQUESTED** [review]

**Review by hskim881028 — APPROVED** [review]

**Review by sonohoshi — APPROVED** [review]

**Pull request merged**